modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/util/CompositeSubscriptionTest.java

```diff
@@ -157,28 +157,27 @@ private void doPublishSubscribe(
 
         lsnr.reset(requested);
 
-        CompletableFuture.runAsync(() -> compPublisher.subscribe(new Subscriber<>() {
-                    @Override
-                    public void onSubscribe(Subscription subscription) {
-                        subscriptionRef.set(subscription);
-                    }
-
-                    @Override
-                    public void onNext(Integer item) {
-                        lsnr.onNext(item);
-                    }
-
-                    @Override
-                    public void onError(Throwable t) {
-                        assert false;
-                    }
-
-                    @Override
-                    public void onComplete() {
-                        lsnr.onComplete();
-                    }
-                })
-        ).join();
+        compPublisher.subscribe(new Subscriber<>() {
```

Comment:
please return back runAsync here, it requeres for the test

Reply:
I think all the guaranties are preserved.
Test thread will continue work right after subscribe method will be finished in background thread.